core/templates/dev/head/pages/exploration_player/HintManagerService.js

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility service for Hints in the learner's view.
+ */
+
+oppia.factory('HintManagerService', [
+  '$timeout', 'playerTranscriptService', 'DELAY_FOR_HINT_FEEDBACK_MSEC',
+  'HINT_REQUEST_STRINGS_ARRAY', 'WAIT_FOR_HINT_MSEC',
+  function(
+      $timeout, playerTranscriptService, DELAY_FOR_HINT_FEEDBACK_MSEC,
+      HINT_REQUEST_STRINGS_ARRAY, WAIT_FOR_HINT_MSEC) {
+    var currentHintIsUsable = false;
+    var numHintsConsumed = 0;
+    var timeout = null;
+
+    return {
+      consumeHint: function() {
+        numHintsConsumed += 1;
+        this.setCurrentHintUsable(false);
+      },
+      getNumHintsConsumed: function() {
+        return numHintsConsumed;
+      },
+      setCurrentHintUsable: function(value) {
+        currentHintIsUsable = value;
+      },
+      isCurrentHintUsable: function() {
```

Comment:
Consider isHintAvailable and similar naming changes elsewhere. To "use" a hint means to apply it in solving a problem, I think, whereas here you're storing whether the hint can be requested or not.

Also let's standardize terminology and use the following constructs throughout:
- isHintAvailable: for whether a hint can be requested right now -- this may fail due to the delay timeout not having elapsed yet, or due to there no longer being any hints to request
- areAllHintsExhausted: for whether there are no longer any hints to request